integration-tests/grpc-hibernate-reactive/src/main/java/com/example/reactive/ReactiveService.java

```diff
@@ -0,0 +1,66 @@
+package com.example.reactive;
+
+import java.lang.reflect.Field;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ManagedContext;
+import io.quarkus.grpc.GrpcService;
+import io.quarkus.hibernate.reactive.panache.Panache;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.operators.multi.processors.BroadcastProcessor;
+
+@GrpcService
+public class ReactiveService implements ReactiveTest {
+
+    private final Set<Integer> requestContexts = Collections.newSetFromMap(new ConcurrentHashMap<>());
+
+    BroadcastProcessor<Item> broadcast = BroadcastProcessor.create();
+
+    private void addContext() {
+        if (!requestContexts.add(requestContextId())) {
+            throw new RuntimeException("request context reused from a different call");
```

Comment:
This may fail intermittently, because `identityHashCode` is not guaranteed to be unique. It's not terribly likely, and I can't think of a better way, but still...

Well maybe, since this is a test, we could use an identity set (`Collections.newSetFromMap(new IdentityHashMap<>())`) and put the `ConcurrentMap` instances into it? :-) (Instead of hash codes of the `ConcurrentMap` instances.)

Reply:
the other integration test, with normal hibernate, uses a better way of testing this, a value from a request scope bean. I thought I changed t his one too, will change it.

Reply:
Ayea, that's even better!

Reply:
done